compiler/rustc_target/src/spec/mod.rs

```diff
@@ -1159,6 +1201,7 @@ pub struct TargetOptions {
     /// Default linker flavor used if `-C linker-flavor` or `-C linker` are not passed
     /// on the command line. Defaults to `LinkerFlavor::Gcc`.
     pub linker_flavor: LinkerFlavor,
+    linker_flavor_json: LinkerFlavorCli,
```

Comment:
I assume all the `_json` fields are named with this suffix as they refer to the json target specs ? Maybe it would be nice if they were named `_cli` to match the CLI type ?

The naming is not super important, especially as I believe these fields are going to be temporary and refactored away soon enough.

Reply:
Yes, these fields are what is parsed directly from json specs.
They use the same `LinkerFlavorCli` structure as command line options for simplicity, but they are semantically json.